lib/diaspora_federation/validators/relayable_validator.rb

```diff
@@ -0,0 +1,11 @@
+module DiasporaFederation
+  module Validators
+    module RelayableValidator
+      def self.included(model)
+        model.class_eval do
+          rule :parent_guid, :guid
```

Comment:
It looks like ```author_signature``` never gets checked really. The only thing that works for the relayable ancestors is ```parent_author_signature```.